core/app/models/spree/inventory_unit.rb

```diff
@@ -62,13 +62,9 @@ class InventoryUnit < Spree::Base
     # @return [Array<Spree::InventoryUnit>] an array of backordered inventory
     #   units for the given stock item
     def self.backordered_for_stock_item(stock_item)
-      # This was refactored from a simpler query because the previous
-      # implementation led to issues once users tried to modify the objects
-      # returned. That's due to ActiveRecord `joins(shipment: :stock_location)`
-      # only returning readonly objects
-      backordered_per_variant(stock_item).select do |unit|
-        unit.shipment.stock_location == stock_item.stock_location
-      end
+      backordered_per_variant(stock_item)
+        .where("spree_shipments.stock_location_id = ?",
```

Comment:
:+1: Now that this doesnt return an array, but a relation, we could probably move it into a scope.

Might be cleaned up by `where(shipment: { stock_location_id: stock_item.stock_location_id })`